test/e2e/resilience/resilience_multizone_k8s.go

```diff
@@ -0,0 +1,273 @@
+package resilience
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/config/core"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+)
+
+func ResilienceMultizoneK8s() {
+	var global, zone1 *K8sCluster
+	var optsGlobal, optsZone1 = KumaK8sDeployOpts, KumaK8sDeployOpts
+	defer GinkgoRecover()
+	Skip("Skipped until https://github.com/kumahq/kuma/issues/1001 is fixed and this tests run reliably")
+
+	BeforeEach(func() {
+		clusters, err := NewK8sClusters(
+			[]string{Kuma1, Kuma2},
+			Silent)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Global
+		global = clusters.GetCluster(Kuma1).(*K8sCluster)
+		err = NewClusterSetup().
+			Install(Kuma(core.Global, optsGlobal...)).
+			Setup(global)
+		Expect(err).ToNot(HaveOccurred())
+		err = global.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		globalCP := global.GetKuma()
+
+		// Cluster 1
+		zone1 = clusters.GetCluster(Kuma2).(*K8sCluster)
+		optsZone1 = append(optsZone1, WithGlobalAddress(globalCP.GetKDSServerAddress()))
+
+		err = NewClusterSetup().
+			Install(Kuma(core.Zone, optsZone1...)).
+			Install(NamespaceWithSidecarInjection(TestNamespace)).
+			Setup(zone1)
+		Expect(err).ToNot(HaveOccurred())
+		err = zone1.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	E2EAfterEach(func() {
+		err := zone1.DeleteKuma(optsZone1...)
+		Expect(err).ToNot(HaveOccurred())
+		err = zone1.DismissCluster()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = global.DeleteKuma(optsGlobal...)
+		Expect(err).ToNot(HaveOccurred())
+		err = global.DismissCluster()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should see global entities in zone after a zone restart", func() {
+		// Create a mesh
+		err := YamlK8s(`
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: my-resilience-mesh-before
+`)(global)
+		Expect(err).ToNot(HaveOccurred())
+
+		// The mesh should make it to zone
+		Eventually(func() string {
+			output, err2 := zone1.GetKumactlOptions().RunKumactlAndGetOutput("get", "meshes")
```

Comment:
s/err2/err/

Also; would it work to just check that `kumactl get mesh my-resilience-mesh-before` succeeds?

Reply:
Yes maybe it would. Sometime we check multiple meshes at once but we can do multiple get anyway